File: carelog-be/src/main/java/carelog/carelog/auth/app/RedisBlacklistService.java
package carelog.carelog.auth.app;

import lombok.*;
import lombok.extern.slf4j.*;
import org.springframework.data.redis.core.*;
import org.springframework.stereotype.*;

import java.time.*;

@Slf4j
@Service
@RequiredArgsConstructor
public class RedisBlacklistService {

    private final StringRedisTemplate redisTemplate;

    public void addToBlacklist(String token, Duration ttl) {
        String key = "blacklist:" + token;
        redisTemplate.opsForValue().set(key, "1", ttl);
        log.info("Blacklist 등록 완료 - key length: {}, ttl: {}", key.length(), ttl);
    }
}

File: carelog-gateway/src/main/kotlin/carelog/gateway/blacklist/RedisBlacklistService.kt
package carelog.gateway.blacklist

import org.springframework.data.redis.core.ReactiveStringRedisTemplate
import org.springframework.stereotype.Service
import reactor.core.publisher.Mono

@Service
class RedisBlacklistService(
    private val redisTemplate: ReactiveStringRedisTemplate
) {
    // carelog-be에서 로그아웃 시 "blacklist:{token}" 키로 Redis에 저장함
    // 여기서는 해당 키 존재 여부만 확인 — true면 블랙리스트된 토큰
    fun isBlacklisted(token: String): Mono<Boolean> =
        redisTemplate.hasKey("blacklist:$token")
}